monix-eval/shared/src/main/scala/monix/eval/internal/TaskRunLoop.scala

```diff
@@ -212,9 +212,17 @@ private[eval] object TaskRunLoop {
         context.frameRef.reset()
 
         // Transporting the current context if localContextPropagation == true.
-        Local.bind(savedLocals) {
+        var prevLocals: Local.Context = null
+        if (savedLocals != null) {
```

Comment:
Is it needed if there is already a null check in `Local.bind`? Or is it just to inline?

Reply:
I'm doing this for inlining, since `Local.bind` is no longer a macro and I don't want to have the penalty of that call in case locals are not disabled.